internal/pkg/cli/deploy/env.go

```diff
@@ -49,11 +58,32 @@ func (d *envDeployer) UploadArtifacts() (map[string]string, error) {
 		return nil, err
 	}
 
+	if d.uploadCustomResourceFlag {
+		return d.uploadCustomResources(resources.S3Bucket)
+	}
+	return d.legacyUploadCustomResources(resources.S3Bucket)
+}
+
+func (d *envDeployer) legacyUploadCustomResources(bucket string) (map[string]string, error) {
 	urls, err := d.uploader.UploadEnvironmentCustomResources(func(key string, objects ...s3.NamedBinary) (string, error) {
-		return d.s3.ZipAndUpload(resources.S3Bucket, key, objects...)
+		return d.s3.ZipAndUpload(bucket, key, objects...)
 	})
 	if err != nil {
-		return nil, fmt.Errorf("upload custom resources to bucket %s: %w", resources.S3Bucket, err)
+		return nil, fmt.Errorf("upload custom resources to bucket %s: %w", bucket, err)
+	}
+	return urls, nil
+}
+
+func (d *envDeployer) uploadCustomResources(bucket string) (map[string]string, error) {
```

Comment:
Why do we need this new `uploadCustomResources` 🤔

Reply:
For environments? they're basically the same except that the functions upload the custom resources in different paths.